WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m

```diff
@@ -0,0 +1,232 @@
+{
+    [super viewDidLoad];
+        
+    self.stackView.layoutMarginsRelativeArrangement = YES;
+    UIEdgeInsets margin = [Menu viewDefaultDesignInsets];
+    margin.top = 0;
+    margin.bottom = 0;
+    self.stackView.layoutMargins = margin;
+    self.stackView.spacing = 4.0;
+    
+    [self setupTextField];
+    [self setupTextFieldDesignViews];
+    [self setupDoneButton];
+    [self setupTrashButton];
+    
+    [self updateTextFieldDesignIconPositioning];
+}
+
+- (void)setupTextField
+{
+    UITextField *textField = self.textField;
+    textField.placeholder = NSLocalizedString(@"Menu Name", @"Menus placeholder text for the name field of a menu with no name.");
+    textField.textColor = [WPStyleGuide darkGrey];
+    [textField addTarget:self action:@selector(hideTextFieldKeyboard) forControlEvents:UIControlEventEditingDidEndOnExit];
+    [textField addTarget:self action:@selector(textFieldValueChanged:) forControlEvents:UIControlEventEditingChanged];
+}
+
+- (void)setupDoneButton
+{
+    UIButton *doneButton = self.doneButton;
+    [doneButton setTitle:NSLocalizedString(@"Done", @"Menu button title for finishing editing the Menu name.") forState:UIControlStateNormal];
+    [doneButton setTitleColor:[WPStyleGuide darkBlue] forState:UIControlStateNormal];
+    doneButton.alpha = 0.0;
+    [doneButton addTarget:self action:@selector(doneButtonPressed) forControlEvents:UIControlEventTouchUpInside];
+}
+
+- (void)setupTrashButton
+{
+    UIButton *trashButton = self.trashButton;
+    [trashButton setTitle:nil forState:UIControlStateNormal];
+    trashButton.tintColor = [WPStyleGuide grey];
+    [trashButton setImage:[Gridicon iconOfType:GridiconTypeTrash] forState:UIControlStateNormal];
+    [trashButton addTarget:self action:@selector(trashButtonPressed) forControlEvents:UIControlEventTouchUpInside];
+    trashButton.backgroundColor = [UIColor clearColor];
+    trashButton.adjustsImageWhenHighlighted = YES;
+}
+
+- (void)setupTextFieldDesignViews
+{    
+    UIView *textFieldDesignView = self.textFieldDesignView;
+    textFieldDesignView.layer.cornerRadius = MenusDesignDefaultCornerRadius;
+    
+    UIImage *image = [Gridicon iconOfType:GridiconTypePencil];
+    UIImageView *imageView = [[UIImageView alloc] initWithImage:image];
+    imageView.translatesAutoresizingMaskIntoConstraints = NO;
+    imageView.tintColor = [WPStyleGuide grey];
+    imageView.contentMode = UIViewContentModeScaleAspectFit;
+    _textFieldDesignIcon = imageView;
+    
+    [textFieldDesignView addSubview:imageView];
+    
+    NSLayoutConstraint *leadingConstraint = [imageView.leadingAnchor constraintEqualToAnchor:self.textField.leadingAnchor];
+    _textFieldDesignIconLeadingConstraint = leadingConstraint;
+    [NSLayoutConstraint activateConstraints:@[
+                                              [imageView.widthAnchor constraintEqualToConstant:14],
+                                              [imageView.heightAnchor constraintEqualToConstant:14],
```

Comment:
really nitpicky but maybe for type consistency could we make these a float: `14.0`?  Also const :)